compat/ioapi.c

```
@@ -108,8 +139,21 @@ static int32_t mz_stream_ioapi_write(void *stream, const void *buf, int32_t size
     } else
         return MZ_PARAM_ERROR;
 
-    written = (int32_t)zwrite(opaque, ioapi->handle, buf, size);
-    return written;
+    //Because zwrite expects unsigned long in size which differs on different system
+    //Need to clamp existed data and feed it in chuncks
+    while (total_written < size)
+    {
+        to_write = (size - total_written > ULONG_MAX) ? ULONG_MAX : (unsigned long) (size - to_write);
```

Cppcheck (2.21.0)

[error] Uninitialized variable

(uninitvar)